File: Lua/Pseudocode/5 Problems/AddTwoNums.lua
--Take two numbers from user, add together, return result
function twoNumberSum()
    -- variables
    local num1 = nil
    local num2 = nil
    local sum = 0
    -- Get the first number
    while num1 == nil or num1 == "" or not tonumber(num1) do
        print("What is your first number: ")
        num1 = io.read()
    end
    -- Get the second number
    while num2 == nil or num2 == "" or not tonumber(num2) do
        print("What is your second number: ")
        num2 = io.read()
    end

    --Add the two numbers and store them in sum
    sum = num1 + num2
    -- Return the sum of the two numbers
    return sum
end

--Take two numbers from user, add together, print result
function printTwoNumberSum()
    -- Print the sum of two numbers
    print(twoNumberSum())
end

File: Python/Pseudocode/5 Problems/AddTwoNums.py
# Have user input two numbers, add them together and print the result.

# Is the input a number
def IsNum(userInput):
    isFloat = False
    try:
        float(userInput)
        isFloat = True
    except:
        isFloat = False

    return isFloat
# def IsNum(userInput):
#    return bool(userInput.isdigit() or isinstance(userInput, float))

# While not a number, keep promting
def TryAgain(userInput):
    while not IsNum(userInput):
        userInput = input("That is not a number, please try again: ")
    return userInput

num1 = input("Enter a number: ")
num1 = TryAgain(num1)
num2 = input("Enter another number: ")
num2 = TryAgain(num2)

print(num1+" + "+num2+" = "+str(float(num1) + float(num2)))